driver/image_adjust_saturation_driver.hpp

```diff
@@ -0,0 +1,378 @@
+          n_stride,
+          c_stride,
+          N,
+          saturation_factor,
+          bound);
+}
+
+template <typename Tgpu, typename Tref>
+class ImageAdjustSaturationDriver : public Driver
+{
+public:
+    ImageAdjustSaturationDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputTensorDesc);
+        miopenCreateTensorDescriptor(&outputTensorDesc);
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyForward() override;
+    int VerifyBackward() override;
+    ~ImageAdjustSaturationDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputTensorDesc);
+        miopenDestroyTensorDescriptor(outputTensorDesc);
+    }
+
+private:
+    InputFlags inflags;
+    int forw;
+
+    miopenTensorDescriptor_t inputTensorDesc;
+    miopenTensorDescriptor_t outputTensorDesc;
+
+    std::unique_ptr<GPUMem> input_gpu;
+    std::unique_ptr<GPUMem> output_gpu;
+    std::unique_ptr<GPUMem> workspace_gpu;
+
+    std::vector<Tgpu> input_host;
+    std::vector<Tgpu> output_host;
+    std::vector<Tref> output_ref;
+
+    float saturation_factor;
+};
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustSaturationDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only the forward pass", "int");
+
+    inflags.AddTensorFlag("input", 'I', "1x3x96x96", "Input Tensor Size");
+    inflags.AddInputFlag("contiguous", 'Z', "1", "Use Contiguous Tensors", "int");
+    inflags.AddInputFlag("saturation", 'S', "0.5", "Saturation", "double");
+
+    inflags.AddInputFlag("iter", 'i', "10", "Number of iterations", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustSaturationDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    saturation_factor = inflags.GetValueDouble("saturation");
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustSaturationDriver<Tgpu, Tref>::GetandSetData()
+{
+    TensorParameters input_vec = inflags.GetValueTensor("input");
+    assert(input_vec.lengths.size() == 4 || input_vec.lengths.size() == 3);
+    if(input_vec.lengths.size() == 3)
+    {
+        // n=1
```

Comment:
Please remove comment

Reply:
Changed the comment to better reflect why an insertion of `n=1` is necessary